### ISSUE TITLE
Character menu doesn't display properly during CGs

### ISSUE THREAD

**Issue by drojf** [07th-mod/umineko-question#87]
Sometimes the character menu bugs out if accessed when a CG is on-screen. I couldn't reproduce on Chiru but supposedly it happens.

A user specifically pointed to the bathtub CG, but I haven't checked that yet.

![cgbug](https://user-images.githubusercontent.com/1249449/36933223-7414a608-1f29-11e8-99d7-ed9fbb71f1f7.png)

Also, could just look through the CGs for this type of attached image (there aren't that many from memory)

edit: Definitely happens at this scene for the Question arcs
```
langjp「そこまでは言わんよ。@無学な女など、話していても疲れるだけだ。@…だが、あまり頭が良過ぎる女はもっと疲れる。@お前のように、自己主張の激しい女では、未来の夫もさぞや苦労すると思うのだがね。」\
langen:dwave 0, kla_3e7:^"I wouldn't go that far.^@:dwave 0, kla_3e8:^  Talking to an unschooled woman is tiring and nothing more.^@:dwave 0, kla_3e9:^  ...However, women are even more tiring when they're too smart.^@:dwave 0, kla_3e10:^  I do believe that a woman as violently assertive as you would bring no end of hardship to her future husband."^\

```

**Comment by drojf** [comment]
I think i've tracked this down to being due to the `set_oldnew_bg ` function, thanks to this scene/report in this (spoiler): https://github.com/07th-mod/umineko-question/issues/112 . The character portraits also get messed up in this instance.

The function is used to display CG's ontop of the existing background, without clearing the existing background (i think), and also set different backgrounds depending on if 'old sprites' mode is enabled

I noticed that after calling `set_oldnew_bg g0103_a,blood_1a,1,22` for example, no sprites will be displayed (using the `ld` command). This does not become 'fixed' until you call `clr_oldnew_bg`. This resulted in a sprite never being displayed in 'new sprites' mode, so I wouldn't be surprised if it also caused the character portraits to go wacky.

If anyone wants to look into it, search for `*set_oldnew_bg` in the script file.

**Comment by drojf** [comment]
We also had this report:

And also another bug I encountered was that the Tips UI during non-background CGs (Dedicated CGs to a scene) was hard to operate
The portraits for the characters in the scheme were black
**And the cursor position didn't actually relate to the object I wanted to click**

I had a report related to cursor position offset before, but they didn't mention it was related to CGs, so I think it's another symptom of this issue,

**Comment by drojf** [comment]
I have closed this in https://github.com/07th-mod/umineko-question/commit/5379e11979630e95888dacb56d7f8d36fd63b9fa . The problem was the method used to hide sprites while CGs were displayed.

- To hide sprites while CGs were displayed, `%hide_new_sprites` was set to 1
- This causes various functions to display a dummy blank sprite when enabled, which does indeed hide the character sprites during CGs
- However, the code which draws the character portraits relies on a function which does this sprite hiding, so the character portraits would also be hidden. (The character portrait code calls `*lspnewold_trans`, which is affected by `%hide_new_sprites`)

I fixed it by not hiding sprites in `*lspnewold_trans`, if you are in the right-click menu (I made a new variable which tracks whether you are in the right-click menu)

I don't think this affects the Answer Arcs - in that script, the only place where `%hide_new_sprites` is set to 1 is the `*set_oldnew_bg`, which is never called.
